test/parallel/test-timers-clearImmediate.js

```diff
@@ -2,17 +2,39 @@
 require('../common');
 const assert = require('assert');
 
-const N = 3;
-let count = 0;
-function next() {
-  const immediate = setImmediate(function() {
-    clearImmediate(immediate);
-    ++count;
+// run clearImmediate in the callback.
+{
+  const N = 3;
+  let count = 0;
+  function next() {
+    const immediate = setImmediate(function() {
```

Comment:
Perhaps we can drop `count` entirely and just wrap the callback in a `common.mustCall()`?